### ISSUE TITLE
M5 -  Consumption Reporting API: Uplift to TS 26.512 V17.7.0

### ISSUE THREAD

**Issue by rjb1000** [5G-MAG/rt-5gms-application-function#114]
# Background

The following changes to TS 26.512 were agreed at SA4#126 (Chicago) and approved at SA#102 (Edinburgh):

* CR0054 "Essential maintenance" in [S4-231631](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_126_Chicago/Docs/S4-231631.zip) and plenary CR Pack [SP-231368](https://www.3gpp.org/ftp/TSG_SA/TSG_SA/TSGS_102_Edinburgh_2023-12/Docs/SP-231368.zip).

The updated specification will be published shortly as TS 26.512 V17.7.0.

This issue captures the changes that need to be made to the reference implementation of the 5GMS AF Consumption Reporting API at reference point M5.

# Summary of changes

1. _ConsumptionReportingUnit_ in clauses 11.3.3.2 and C.4.2 has new properties _server‌Endpoint‌Address_ and _client‌Endpoint‌Address_ that replace the old property _endpointAddress_.

The OpenAPI YAML at [TS26512_M5_ConsumptionReporting.yaml](https://forge.3gpp.org/rep/all/5G_APIs/-/blob/2dcd1b3de52ffd10c61dbc895b6f44c8b5b5082b/TS26512_M5_ConsumptionReporting.yaml) should be used to implement this in preference to the one on 3GPP Forge because...

**Comment by devbbc** [comment]
The changes to the _ConsumptionReportingUnit_ are handled by the template file [TS26512_M5_ConsumptionReporting.yaml](https://forge.3gpp.org/rep/all/5G_APIs/-/blob/2dcd1b3de52ffd10c61dbc895b6f44c8b5b5082b/TS26512_M5_ConsumptionReporting.yaml).

No changes to AF code required.

**Comment by rjb1000** [comment]
> No changes to AF code required.

Is this because Consumption Reports are simply stored verbatim to disk in the current implementation?

If so, we can probably delete this issue.

**Comment by devbbc** [comment]
Yes, the 5GMS AF stores the Consumption Reports in its internal data structure and retrieves it.
However, the AF currently does not process nor de-reference the elements in Consumption Report data structure.
